--- applications/adjoint_tests/working/psyad/kernel/core_dynamics/tl_project_eos_pressure_kernel_mod.F90.orig	2024-07-06 10:14:11.311523209 +0100
+++ applications/adjoint_tests/working/psyad/kernel/core_dynamics/tl_project_eos_pressure_kernel_mod.F90	2024-07-06 10:15:52.481153428 +0100
@@ -112,7 +112,7 @@
                                  nqp_h, nqp_v, wqp_h, wqp_v                    &
                                  )
 
-  use tl_calc_exner_pointwise_mod, only: tl_calc_exner_pointwise
+  ! use tl_calc_exner_pointwise_mod, only: tl_calc_exner_pointwise
   use sci_coordinate_jacobian_mod, only: coordinate_jacobian
 
   implicit none
@@ -160,9 +160,13 @@
   real(kind=r_def), dimension(3,3,nqp_h,nqp_v) :: jac
 
   real(kind=r_def) :: exner_at_quad, rho_at_quad, theta_vd_at_quad
-  real(kind=r_def) :: exner_eos
   real(kind=r_def) :: ls_rho_at_quad, ls_theta_vd_at_quad
 
+  real(kind=r_def) :: tmp_ls_exner, tmp_exner
+
+  !RF need to add this to adjoint: use planet_config_mod, only : kappa, Rd, p_zero
+  real(kind=r_def) :: kappa, Rd, p_zero
+
   ipanel = int(panel_id(map_pid(1)), i_def)
 
   do k = 0, nlayers-1
@@ -219,9 +223,14 @@
         end do
 
         ! Calculation
-        call tl_calc_exner_pointwise( exner_eos, rho_at_quad, theta_vd_at_quad, &
-                                      ls_rho_at_quad, ls_theta_vd_at_quad )
-        exner_at_quad = wqp_h(qp1) * wqp_v(qp2) * dj(qp1,qp2) * exner_eos
+
+        tmp_ls_exner = ( ( Rd / p_zero ) * ls_rho_at_quad * ls_theta_vd_at_quad ) ** &
+               ( kappa / ( 1.0_r_def - kappa ) )
+
+        tmp_exner = ( kappa / ( 1.0_r_def - kappa ) ) * tmp_ls_exner * &
+             ( ( rho_at_quad / ls_rho_at_quad ) + ( theta_vd_at_quad / ls_theta_vd_at_quad )  )
+
+        exner_at_quad = wqp_h(qp1)*wqp_v(qp2)*dj(qp1,qp2)*tmp_exner
 
         do df = 1, ndf_w3
           r_exner(df) = r_exner(df) + w3_basis(1,df,qp1,qp2)*exner_at_quad
